tasks/src/test/java/webdriver/google_cloud/page/TenMinuteEmailHomePage.java

```diff
@@ -1,40 +1,61 @@
 package webdriver.google_cloud.page;
 
-import static webdriver.google_cloud.browser.Browser.waitForVisibleElement;
+import static webdriver.google_cloud.browser.Browser.waitElementToBeClickable;
+import static webdriver.google_cloud.browser.Browser.waitForVisibility;
+import static webdriver.google_cloud.util.StringUtils.EMAIL_PAGE_URL;
 
-import org.openqa.selenium.By;
+import java.util.concurrent.TimeUnit;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.openqa.selenium.Keys;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.FindBy;
-import org.openqa.selenium.support.ui.ExpectedConditions;
-import org.openqa.selenium.support.ui.WebDriverWait;
 
 public class TenMinuteEmailHomePage extends GoogleCloudAbstractPage {
-  @FindBy(id = "mailAddress")
+  private final Logger logger = LogManager.getRootLogger();
+
+  @FindBy(css = "button.click-to-copy")
+  private WebElement mailButton;
+  @FindBy(id = "mail")
   private WebElement mailInput;
-  @FindBy(xpath = "//*[@role='tab']")
-  private WebElement messagesList;
+  @FindBy(css = "input.emailbox-input")
+  private WebElement mailBox;
+
+  @FindBy(xpath = "//li[not(@class='hide ')]//*[@class='arrow-link-ico']")
+  private WebElement messageLink;
   @FindBy(xpath = "//*[@id='mobilepadding']//h2")
-  private WebElement totalCostFromEmail;
+  private WebElement totalCostFromEmailLabel;
 
   public TenMinuteEmailHomePage(WebDriver driver) {
     super(driver);
   }
 
-  public String getTotalCostFromEmailText() {
-    waitForVisibleElement(driver, totalCostFromEmail);
-    return totalCostFromEmail.getText();
+  public TenMinuteEmailHomePage openPage() {
+    driver.get(EMAIL_PAGE_URL);
+    driver.manage().timeouts().pageLoadTimeout(30, TimeUnit.SECONDS);
+    logger.info("Page " + EMAIL_PAGE_URL + " was opened");
+    return this;
+  }
+
+  public TenMinuteEmailHomePage clickOnMail() {
+    waitElementToBeClickable(driver, mailInput).click();
+    return this;
   }
 
   public void copyAddress() {
-    waitForVisibleElement(driver, mailInput);
-    mailInput.sendKeys(Keys.chord(Keys.CONTROL, "c"));
+    waitForVisibility(driver, mailInput).sendKeys(Keys.chord(Keys.CONTROL, "a"));
+    waitForVisibility(driver, mailInput).sendKeys(Keys.chord(Keys.CONTROL, "c"));
+    logger.info("Generated mail was copied");
   }
 
   public void viewEmail() {
-    new WebDriverWait(driver, 100)
-        .until(ExpectedConditions.elementToBeClickable(By.xpath("//*[@role='tab']")));
-    messagesList.click();
+    waitForVisibility(driver, messageLink);
+    waitElementToBeClickable(driver, messageLink).click();
```

Comment:
Looks like the native method checks for an element's visibility inside, so you don't neet to do it twice
org.openqa.selenium.support.ui.ExpectedConditions#elementToBeClickable(org.openqa.selenium.WebElement)